design_drafts/transmission_loading_and_parametrization.md

```diff
@@ -0,0 +1,194 @@
+
+The value transformation can be responsible for something as simple as a reducer transmission, where a certain *joint-level* value is divided by a factor accounting for a mechanical reduction rate to compute the *actuator-level* command to be sent to the actuator while the inverse is applied for feedback from the actuator.
+
+
+![ros2_control Simple Transmission][simple_transmission]
+
+
+More complex mechanical transmissions involve more than one joint where per-joint reduction rate depends on the state of other joints in the system, such as a four-bar linkage  transmission.
+
+![ros2_control FourBarLinkage Transmission][four_bar_linkage_transmission]
+
+Disclaimer: The `transmission_interface` library is not always required. It mainly depends on the level of abstraction the hardware implements and makes available through the hardware-level communication interfaces. An industrial arm for instance will typically expose *joint-level* interfaces through their comms and will have individually actuated joints. A humanoid robot on the other hand may employ differential transmissions for implementing multi-axis joints often seen in wrists, four-bar-linkages for larger parts of the body, such as torso movement, or a complex leg with a single actuator but multiple joints with feedback from them, etc.
+
+
+## How it works in ROS Noetic and problems
+
+To facilitate extendability, transmissions are implemented as plugins that are loaded and configured at startup time. This allows anyone to implement their custom transmission plugins in any ROS package without having to change `ros_control`.
+
+### XML format
+
+An additional XML description for each joint has to be added to the URDF in order to define transmissions on them.
+
+```
+<?xml version="1.0"?>
+
+<robot name="robot" xmlns="http://www.ros.org">
+
+  <transmission name="four_bar_linkage_trans">
+    <type>transmission_interface/FourBarLinkageTransmission</type>
+    <joint name="bar_joint">
+      <role>joint2</role>
+      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
+    </joint>
+    <joint name="foo_joint">
+      <role>joint1</role>
+      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
+    </joint>
+    <actuator name="foo_actuator">
+      <role>actuator1</role>
+      <mechanicalReduction>50</mechanicalReduction>
+    </actuator>
+    <actuator name="bar_actuator">
+      <role>actuator2</role>
+      <mechanicalReduction>-50</mechanicalReduction>
+    </actuator>
+  </transmission>
+
+</robot>
+```
+
+Let's dissect the XML above:
+
+* `<type>transmission_interface/FourBarLinkageTransmission</type>` defines the transmission plugin to be loaded, parameters are following
+* `<joint name="bar_joint">` defines the role and hardware interface of joints. The joint `bar_joint` should already be defined in the URDF.
+* `<role>joint2</role>` is internal to transmission implementations, we'll not explain that here
```

Comment:
Since below we talk about making `role` a mandatory tag, then maybe it is worth to try and explain what is being used for.

Reply:
I agree on that. I am guessing the `role` name is internal to ros2_control transmission: to provide a key-name in an std::map to create a relationship  between a joint and and actuator?

Reply:
Yes they are, thing of roles here as axes at very specific points in a transmission.
four bar linkage is a good example: https://en.wikipedia.org/wiki/Four-bar_linkage#/media/File:Linkage_four_bar_fixed.svg